### PR TITLE
Issue 2 viewing the entries

### PR DESCRIPTION
**Pull request by dcatan89**
Screen Shot 2021-12-26 at 4 29 23 PM
Screen Shot 2021-12-26 at 4 35 37 PM
Screen Shot 2021-12-26 at 4 35 56 PM
Screen Shot 2021-12-26 at 4 36 13 PM

### CODE REVIEW

**Comment by dcatan89** [comment]
issue2

File: js/main.js [hunk 1]
```diff
@@ -27,6 +27,110 @@ function submitHandler(event) {
...
+      $view[i].classList.add('hidden');
+    }
+  }
+  for (i = 0; i < $view.length; i++) {
+    if ($view[i].getAttribute('data-view') === string && data.view === 'entry-form') {
+      $view[i].classList.remove('hidden');
+    } else {
+      $entriesAnchor.className = 'entries-anchor view';
+    }
+  }
```
Comment (codingcodymiller) [inline js/main.js]: You can reduce all this to the following:
```
  for (var i = 0; i < $view.length; i++) {
    if ($view[i].getAttribute('data-view') === string) {
      $view[i].classList.remove('hidden');
    } else {
      $view[i].classList.add('hidden');
    }
  }
```

Comment (codingcodymiller) [reply js/main.js]: Examine this and make sure you understand how it works, please ask me any questions if you have them.

File: js/main.js [hunk 2]
```diff
@@ -27,6 +27,110 @@ function submitHandler(event) {
...
+
+window.addEventListener('DOMContentLoaded', domContentLoadedHandle);
+
+var $view = document.querySelectorAll('.view');
+var $entriesAnchor = document.querySelector('.entries-anchor');
+var $noEntries = document.querySelector('.no-entries');
+var $newButton = document.querySelector('.new-button');
+
+function viewSwapEntries(event) {
+  event.preventDefault();
```
Comment (codingcodymiller) [inline js/main.js]: this line is not necessary, you only need to prevent default behavior for submit event handlers.

File: js/main.js [hunk 3]
```diff
@@ -27,6 +27,110 @@ function submitHandler(event) {
...
+
+    if ($viewEntryForm === $view[i].getAttribute('data-view')) {
+      $view[i].classList.remove('hidden');
+      dataView($viewEntryForm);
+    } else {
+      $view[i].classList.remove('hidden');
+      $view[i].classList.add('hidden');
+      $entriesAnchor.className = 'entries-anchor view';
+    }
+  }
```
Comment (codingcodymiller) [inline js/main.js]: All of this functionality is already covered in `dataView`. You should just call `dataView` with an argument of the value of the `data-view` attribute on the anchor element that was clicked

File: js/main.js [hunk 4]
```diff
@@ -27,6 +27,110 @@ function submitHandler(event) {
   data.entries.unshift(entryValues);
   submitForm.reset();
   imgOnScreen.setAttribute('src', 'images/placeholder-image-square.jpg');
+  $ulEntries.prepend(generateEntriesDOMTree(entryValues));
```
Comment (codingcodymiller) [inline js/main.js:30]: Does the view change to the entries view when you submit the form? The app should go to the entries view as soon as the new entry is submitted.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by dcatan89** [comment]
issue 2 2

File: js/main.js [hunk 5]
```diff
@@ -27,6 +27,96 @@ function submitHandler(event) {
...
+var $view = document.querySelectorAll('.view');
+var $entriesAnchor = document.querySelector('.entries-anchor');
+var $noEntries = document.querySelector('.no-entries');
+var $newButton = document.querySelector('.new-button');
+
+function viewSwapEntries(event) {
+  if (data.entries.length === 0) {
+    $noEntries.classList.remove('hidden');
+  }
+  var $dataViewEntries = event.target.getAttribute('data-view');
```
Comment (codingcodymiller) [inline js/main.js]: Rename this to `viewName`

File: js/main.js [hunk 6]
```diff
@@ -27,6 +27,96 @@ function submitHandler(event) {
...
+
+var $view = document.querySelectorAll('.view');
+var $entriesAnchor = document.querySelector('.entries-anchor');
+var $noEntries = document.querySelector('.no-entries');
+var $newButton = document.querySelector('.new-button');
+
+function viewSwapEntries(event) {
+  if (data.entries.length === 0) {
+    $noEntries.classList.remove('hidden');
+  }
```
Comment (codingcodymiller) [inline js/main.js]: You can remove this, it's already in `dataView`

File: js/main.js [hunk 7]
```diff
@@ -27,6 +27,96 @@ function submitHandler(event) {
...
+}
+
+window.addEventListener('DOMContentLoaded', domContentLoadedHandle);
+
+var $view = document.querySelectorAll('.view');
+var $entriesAnchor = document.querySelector('.entries-anchor');
+var $noEntries = document.querySelector('.no-entries');
+var $newButton = document.querySelector('.new-button');
+
+function viewSwapEntries(event) {
```
Comment (codingcodymiller) [inline js/main.js]: Rename this to `handleViewSwap`

File: js/main.js [hunk 8]
```diff
@@ -27,6 +27,96 @@ function submitHandler(event) {
...
+      dataView($dataViewEntries);
+    }
+    var $viewEntryForm = event.target.getAttribute('data-view');
+    for (i = 0; i < $view.length; i++) {
+
+      if ($viewEntryForm === $view[i].getAttribute('data-view')) {
+        dataView($viewEntryForm);
+      }
+    }
+  }
```
Comment (codingcodymiller) [inline js/main.js]: You can replace all this with `dataView(viewName)`. All this looping to check the view is already done in `dataView`.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 6 above)
Comment (codingcodymiller) [reply js/main.js]: This still needs to be addressed

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**